### PR TITLE
Make the `Deploy LLMs` template use the new RayLLM API

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
This change makes the `Deploy LLMs` template use the new RayLLM API.

### CODE REVIEW

File: templates/endpoints_v2/README.ipynb [hunk 1]
```diff
@@ -8,35 +8,37 @@
     "\n",
     "**⏱️ Time to complete**: 10 min (20 on GCP)\n",
     "\n",
-    "This guide walks you through how to deploy optimized LLMs in Anyscale through RayLLM. It includes a number of pre-tuned configs for Llama2, Mistral, Mixtral, embedding models, and more in the `models` directory.\n",
+    "This guide walks you through how to deploy LLMs in Anyscale through RayLLM.\n",
```
Comment (kouroshHakha) [inline templates/endpoints_v2/README.ipynb:11]: revise the time? I don't why why for GCP it says it takes longer? in both I think it takes 10 min.

File: templates/endpoints_v2/README.ipynb [hunk 2]
```diff
@@ -8,35 +8,37 @@
...
-    "- How to configure autoscaling and other optimization parameters\n",
-    "\n",
-    "**Note**: This guide is hosted within an Anyscale workspace, which provides easy access to compute resources. Check out the `Introduction to Workspaces` template for more details.\n",
-    "\n",
-    "## Step 1 - Run the model locally in the Workspace\n",
-    "\n",
-    "We provide a starter command to run Llama and Mistral-family models via Ray Serve. You can specify the arguments, such as Lora, GPU type and tensor parallelism via the command. You can also follow the [guide](cookbooks/CustomModels.ipynb) to bring your own models.\n",
-    "\n",
-    "Please note that if you would like to serve a model whose architecture is different from the provided list of models, we highly recommend you manually going over the generated model config file to provide the correct values."
+    "**Note**: This guide is hosted within an Anyscale workspace, which provides easy access to compute resources. Check out the `Introduction to Workspaces` template for more details."
```
Comment (kouroshHakha) [inline templates/endpoints_v2/README.ipynb:13]: Link `Introduction to Workspaces`?

**Review by kouroshHakha — COMMENTED** [review]
I don't like the way the initial readme is setup right now. It doesn't contextualize what I am doing as a user. We need to write the readme from scratch. 

User doesn't know what rayllm is, so a simple blurb is required in the first paragraph.


Something like:
"This template comes with a installed library for serving LLMs on Anyscale called [RayLLM](link to version). It provides the fastest way to try out open-weight LLMs. You can read more about this library and its features in the [docs](link). For learning on how to training or offline for doing batch inference you can refer to the [serving template](https://console.anyscale.com/v2/template-preview/???) or the [offline batch inference template](https://console.anyscale.com/v2/template-preview/batch-llm), respecitvely."


Basically copy / paste the story telling pattern of llmforge's readme. There needs to be some specific details around deploying a service locally vs. service but that's it. Have a what's next section + end-to-end example section with function calling. Merge my PR on function calling with this PR actually.

File: templates/endpoints_v2/README.ipynb (hunk 2 above)
Comment (shrekris-anyscale) [reply templates/endpoints_v2/README.ipynb:13]: I added a link.

File: templates/endpoints_v2/README.ipynb (hunk 1 above)
Comment (shrekris-anyscale) [reply templates/endpoints_v2/README.ipynb:11]: I revised the time.

File: templates/endpoints_v2/end-to-end-examples/function_calling/serve_llama_3p1.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+applications:
+- args:
+    embedding_models: []
+    function_calling_models:
+    - ./llm_configs/meta-llama--Meta-Llama-3_1-8B-Instruct.yaml
+    models: []
+    multiplex_lora_adapters: []
+    multiplex_models: []
+    vllm_base_models: []
+  import_path: aviary_private_endpoints.backend.server.run:router_application
```
Comment (kouroshHakha) [inline templates/endpoints_v2/end-to-end-examples/function_calling/serve_llama_3p1.yaml:10]: use the new name space?

File: templates/endpoints_v2/README.ipynb [hunk 4]
```diff
@@ -6,37 +6,54 @@
...
+    "As well as operational features to efficiently scale LLM apps:\n",
+    "- Optimizations such as continuous batching, quantization and streaming.\n",
+    "- Production-grade autoscaling support, including scale-to-zero.\n",
+    "- Native multi-GPU & multi-node model deployments.\n",
     "\n",
-    "## Step 1 - Run the model locally in the Workspace\n",
+    "This template explains how to set up, run, and query LLMs with RayLLM.\n",
     "\n",
-    "We provide a starter command to run Llama and Mistral-family models via Ray Serve. You can specify the arguments, such as Lora, GPU type and tensor parallelism via the command. You can also follow the [guide](cookbooks/CustomModels.ipynb) to bring your own models.\n",
+    "To learn more about RayLLM, check out [the docs](http://https://docs.anyscale.com/llms/serving/intro). If you're interested in fine-tuning, check out the [fine-tuning template](../fine-tune-llm_v2/README.ipynb).\n",
```
Comment (kouroshHakha) [inline templates/endpoints_v2/README.ipynb:24]: link the actual workspace template for [fine-tuning template] i.e. https://console.anyscale.com/v2/template-preview/finetuning_llms_v2

**Review by kouroshHakha — COMMENTED** [review]
ok nice. Some nits:

File: templates/endpoints_v2/end-to-end-examples/function_calling/serve_llama_3p1.yaml (hunk 3 above)
Comment (shrekris-anyscale) [reply templates/endpoints_v2/end-to-end-examples/function_calling/serve_llama_3p1.yaml:10]: Good catch, I updated this config to use the new API.

File: templates/endpoints_v2/README.ipynb (hunk 4 above)
Comment (shrekris-anyscale) [reply templates/endpoints_v2/README.ipynb:24]: Got it, I updated the template links.

**Review by kouroshHakha — APPROVED** [review]
